lib/PushConstant.cpp

```diff
@@ -85,4 +87,24 @@ Value *GetPushConstantPointer(BasicBlock *BB, PushConstant pc) {
   return Builder.CreateInBoundsGEP(GV, Indices);
 }
 
+bool UsesGlobalPushConstants(Module &M) {
+  return ShouldDeclareEnqueuedLocalSize(M) || ShouldDeclareGlobalOffset(M);
```

Comment:
Presumably this needs to rebased and updated to include the new global push constants added by #572.

Reply:
Do that right now.

Reply:
Done